lib/libsacd/dsdiff.c

```diff
@@ -644,7 +644,7 @@ static size_t dsdiff_write_frame(scarletbook_output_format_t *ft, const uint8_t
 
 #ifdef _WIN32
             handle->frame_indexes[handle->frame_count - 1].offset = _ftelli64(ft->fd) + DST_FRAME_DATA_CHUNK_SIZE;
-#elif defined(__lv2ppu__) || defined(__APPLE__)
+#elif defined(__lv2ppu__) || defined(__APPLE__) || defined(__ANDROID__)
```

Comment:
Not sure if this will be wrong after bumping android API, so the following would be more fail safe.

```suggestion
#elif defined(__lv2ppu__) || defined(__APPLE__) || (defined(__ANDROID_API__) && __ANDROID_API__ < 24)
```

Sadly this only fixes android-aarch64, android-arm also doesn't have ftello (with `-D_FILE_OFFSET_BITS=64` from Kodi). It's the same issue Kodi has with python.

Reply:
uhm, you are contradicting yourself. is android okay now or not? do you mean that your change will only work on aarch64?

Reply:
- android-aarch64 **has** ftello
- android-arm **doesn't have** ftello